src/libwalletqt/Wallet.cpp

```diff
@@ -1021,6 +1037,20 @@ void Wallet::keyReuseMitigation2(bool mitigation)
     m_walletImpl->keyReuseMitigation2(mitigation);
 }
 
+void Wallet::onWalletPassphraseNeeded(bool on_device)
+{
+    emit this->walletPassphraseNeeded(on_device);
+}
+
+void Wallet::onPassphraseEntered(const QString &passphrase, bool enter_on_device, bool entry_abort)
+{
+    PassphraseDelegate * pdelegate = dynamic_cast<PassphraseDelegate*>(m_walletListener);
```

Comment:
`dynamic_cast` is not needed here.

Reply:
without it I cannot use the `PassphraseDelegate`: `Monero::WalletListener *m_walletListener;` 

The `m_walletListener` is instantiated with `class WalletListenerImpl : public  Monero::WalletListener, public PassphraseDelegate` but without dynamic cast I think I cannot reach the `PassphraseDelegate`

Reply:
Do you see reason to not make it `WalletListenerImpl *m_walletListener;`?

Reply:
We would have to move `WalletListenerImpl` out of the `Wallet.cpp`, probably to a separate `.h, .cpp` files.

Reply:
That's okay

Reply:
Done, pls check and resolve when ok.

Reply:
Looks okay to me